### PR TITLE
[WIP] Feature: adds 'publicationHistoryEvent' to articles

### PR DESCRIPTION
**Pull request by lsh-0**
Reverse link: https://github.com/elifesciences/issues/issues/4702

Modelling:

```
<pub-history>
    <event event-type="preprint-publication">
        <event-desc>This article was ogininally published as a <ext-link ext-link-type="uri" 
            xlink:href="https://www.biorxiv.org/content/early/2017/03/24/118356">preprint on Biorxiv</ext-link>
        </event-desc>
        <date iso-8601-date="2016-03-01">
            <day>24</day>
            <month>03</month>
            <year>2017</year>
        </date>  
    </event>
</pub-history> 
```

### CODE REVIEW

**Comment by lsh-0** [comment]
I'm not clear what the distinction is between 'snippet', 'misc' and 'blocks' so I hope the bits are living in the right place.

I also assumed that the ./src/snippets/article.v1.yaml forms the basis of POA/VOR articles and is included in v2 as well.

File: src/snippets/article.v1.yaml
Comment (giorgiosironi) [inline src/snippets/article.v1.yaml]: the snippet is used in all listings so this seems a bit too much detail for that use case.
What about the schema of `/article/:id/versions`:
https://github.com/elifesciences/api-raml/blob/1524e9126fffdc16f2de7549d79666535d5dfe40/src/model/article-history.v1.yaml
or the general body of every `/article/:id` (for both POA and VOR): 
https://github.com/elifesciences/api-raml/blob/1524e9126fffdc16f2de7549d79666535d5dfe40/src/misc/article.v2.yaml

File: src/misc/publicationHistoryEvent.v1.yaml
Comment (giorgiosironi) [inline src/misc/publicationHistoryEvent.v1.yaml]: there are names like `versionDate` and `statusDate` around, so this may be the generic name of those

File: src/misc/publicationHistoryEvent.v1.yaml
Comment (giorgiosironi) [inline src/misc/publicationHistoryEvent.v1.yaml]: theoretically someone could put HTML in here, do we have anything stricter than `string`?

Comment (giorgiosironi) [reply src/misc/publicationHistoryEvent.v1.yaml]: probably not: https://github.com/elifesciences/api-raml/blob/40b376afaa33083a9a86dd3516f171e091e7ffd0/src/model/subject.v1.yaml#L7-L9

File: src/snippets/article.v1.yaml
Comment (gnott) [reply src/snippets/article.v1.yaml]: Looking at `<event>` (https://jats.nlm.nih.gov/archiving/tag-library/1.2d2/element/event.html) and its intent, it allows you to add some additional information about a publication event (or non-publication event). It seems to me more closely aligned with the `accepted` and `received` dates found currently in the `<history>` tag. These dates are associated with all versions of the article and, although the preprint date would not change between versions, it may not be precise to associate a preprint with article itself, and might be better aligned with the `article-history.v1.yaml` somehow? I don't have a concrete example because it is not easy.

File: src/misc/publicationHistoryEvent.v1.yaml
Comment (lsh-0) [reply src/misc/publicationHistoryEvent.v1.yaml]: In the example Melissa gave us of it's intended usage (it's in the PR description), the description contains the external link embedded within the text. I think I argued against this, extracting the URI from an attribute or discrete element elsewhere would have been less ambiguous and the description may contain many such links. I don't think plain or escaped html would get through production, their standards are quite high.

**Comment by lsh-0** [comment]
I've moved the definition inline with the current article history endpoint.

**Comment by gnott** [comment]
I have a good feeling about the structure on this one, great job @lsh-0 !

File: src/model/article-history.v1.yaml
Comment (giorgiosironi) [inline src/model/article-history.v1.yaml]: but this is a [real life event](https://github.com/elifesciences/api-raml/blob/develop/src/samples/event/v2/complete.json), not a state change

Comment (lsh-0) [reply src/model/article-history.v1.yaml]: a state change isn't a type of event? Are you objecting to my use of 'event' (which is fair enough, I didn't know that other event existed). I'm working with very little input here, if you have better ideas then please take a crack at it.

Comment (lsh-0) [reply src/model/article-history.v1.yaml]: I could call it HistoryEvent? or HistoricalEvent??

Comment (gnott) [reply src/model/article-history.v1.yaml]: I like `HistoryEvent`. You wouldn't need a `starts`, `ends` and `published` dates if you defined a new snippet, like the current `event` snippet requires.

Comment (lsh-0) [reply src/model/article-history.v1.yaml]: oh - I see what I've done. This is a typo, that `- $ref: ../snippets/event.v1.yaml` isn't meant to be there.

Comment (lsh-0) [reply src/model/article-history.v1.yaml]: actually it looks all kinds of messed up.

Comment (lsh-0) [reply src/model/article-history.v1.yaml]: ok, just pushed what I intended to commit. it doesn't `node compile` so I'm investigating why

Comment (lsh-0) [reply src/model/article-history.v1.yaml]: fixed. if this is more acceptable I'll write some samples

File: src/model/article-history.v1.yaml
Comment (giorgiosironi) [inline src/model/article-history.v1.yaml:29]: this is more in line with what I meant, not tying this to the other data type of "marketing" events

**Comment by lsh-0** [comment]
May I get some more eyes on this please? There was a bad commit I had to fix up but it looks ok now.

**Review by gnott — APPROVED** [review]
Looks pretty clean and comprehensive to me 👍 Whether the description field is enough to display it correctly on Journal I'm not sure.

**Comment by lsh-0** [comment]
Thanks @gnott 

@thewilkybarkid , if you have the time I'd like an opinion from you. My edits to the spec feel naive.

**Comment by lsh-0** [comment]
I'll be working on this tomorrow, last opportunity to review before I merge

**Comment by thewilkybarkid** [comment]
@lsh-0 Please don't merge this. This feature (ie getting it to appear in Journal) isn't currently in our to do list, and AFAIK we don't have a spec as to what the output would need to be anyhow.

/cc @Maelplaine @BlueReZZ

**Comment by nlisgo** [comment]
@lsh-0 closing. Reopen or create an issue if you want to revisit this.

**Pull request closed without merging**
